server.js

```diff
@@ -24,10 +24,54 @@ app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Possible path/routes to take: /titles, id/:id, titles/:type (like TV Show or Movie) and titles/ search query for title, director or country')
 })
 
+app.get('/titles', (req, res) => {
+  //query parameter
+  const searchString = req.query.search
+
+  let filtredTitles = netflixData
+
+  if (searchString) {
+    //Filter once on multiple fields
+
+    filtredTitles = filtredTitles.filter(item => {
+      const itemTitle = item.title.toString()
+      const itemDirector = item.director.toString()
+      const itemCountry = item.country.toString()
+
+      return itemTitle.toLowerCase().includes(searchString) ||
+        itemDirector.toLowerCase().includes(searchString) ||
+        itemCountry.toLowerCase().includes(searchString)
+    })
+  }
+  res.json(filtredTitles)
+})
+
+
+//Filter out type: TV Show or Movie (correct spelling needed)
+app.get('/titles/:type', (req, res) => {
```

Comment:
It would be nice if this was part of the `/titles` endpoint (and it'd be more RESTful) - then make use of query params to filter on type. So you could have `/titles` to return all netflix titles and then `/titles?type=movie` for movies, etc. Then you'd be able to filter by both type and search for titles at the same time :)